dotnet/src/SemanticKernel/SkillDefinition/SkillCollection.cs

```diff
@@ -38,7 +38,7 @@ public ISkillCollection AddFunction(ISKFunction functionInstance)
         Verify.NotNull(functionInstance, "The function is NULL");
 
         ConcurrentDictionary<string, ISKFunction> skill = this._skillCollection.GetOrAdd(functionInstance.SkillName, static _ => new(StringComparer.OrdinalIgnoreCase));
-        skill.TryAdd(functionInstance.Name, functionInstance);
+        skill.AddOrUpdate(functionInstance.Name, functionInstance, (_, _) => functionInstance);
```

Comment:
If you really want to allow replacement, it can just use the indexer:
```suggestion
        skill[functionInstance.Name] = functionInstance;
```